users/romus/romus.c

```diff
@@ -0,0 +1,633 @@
+\*-------------*/
+#ifdef UNICODE_ENABLE
+#endif
+
+/*-----------------*\
+|*-----SECRETS-----*|
+\*-----------------*/
+// Enabled by adding a non-tracked secrets.h to this dir.
+#if (__has_include("secrets.h"))
+#include "secrets.h"
+#endif
+
+/*---------------*\
+|*-----MUSIC-----*|
+\*---------------*/
+#ifdef AUDIO_ENABLE
+float tone_game[][2]    = SONG(ZELDA_PUZZLE);
+float tone_return[][2]  = SONG(ZELDA_TREASURE);
+float tone_linux[][2]   = SONG(UNICODE_LINUX);
+float tone_windows[][2] = SONG(UNICODE_WINDOWS);
+#endif
+
+/*-------------------*\
+|*-----TAP-DANCE-----*|
+\*-------------------*/
+#ifdef TAP_DANCE_ENABLE
+qk_tap_dance_action_t tap_dance_actions[] = {
+    // Shift on double tap of semicolon
+    [SCL] = ACTION_TAP_DANCE_DOUBLE( KC_SCLN, KC_COLN )
+};
+#endif
+
+/* In keymaps, instead of writing _user functions, write _keymap functions
+ * The __attribute__((weak)) allows for empty definitions here, and during
+ * compilation, if these functions are defined elsewhere, they are written
+ * over. This allows to include custom code from keymaps in the generic code
+ * in this file.
+ */
+__attribute__ ((weak)) void matrix_init_keymap(void) { }
+__attribute__ ((weak)) void matrix_scan_keymap(void) { }
+__attribute__ ((weak)) bool process_record_keymap(uint16_t keycode, keyrecord_t *record) {
+    return true;
+}
+__attribute__ ((weak)) uint32_t layer_state_set_keymap (uint32_t state) {
+    return state;
+}
+__attribute__ ((weak)) void led_set_keymap(uint8_t usb_led) { }
+
+/* ----------------------- *\
+ * -----RGB Functions----- *
+\* ----------------------- */
+#ifdef RGBLIGHT_ENABLE
+// Storage variables
+extern  rgblight_config_t rgblight_config;
+bool    base_sta;   // Keeps track if in saveable state
+bool    base_tog;   // Whether base state is active or not
+int     base_hue;   // Hue value of base state
+int     base_sat;   // Saturation value of base state
+int     base_val;   // Brightness value of base state
+uint8_t base_mod;   // Animation mode of the base state
+
+// Save the current state of the rgb mode
+void rgblight_saveBase(void) {
+    base_hue = rgblight_config.hue;
+    base_sat = rgblight_config.sat;
+    base_val = rgblight_config.val;
+    base_mod = rgblight_config.mode;
+    base_tog = rgblight_config.enable;
+    base_sta = false;   // If saving, that means base layer is being left
+}
+
+// Load the base state back 
+void rgblight_loadBase(void) {
+    // Don't do anything if not enabled
+    if ( !base_sta ) {
+        if ( base_tog ) {
+            rgblight_enable();
+            rgblight_mode( base_mod );
+            rgblight_sethsv( base_hue, base_sat, base_val );
+        } else {
+            rgblight_disable();
+        }
+    }
+    // Mark that base is loaded, and to be saved before leaving
+    base_sta = true;
+}
+
+// Set to plain HSV color
+void rgblight_colorStatic( int hu, int sa, int va ) {
+    // First, it must be enabled or color change is not written
+    rgblight_enable();
```

Comment:
Keep in mind that all of these functions save to EEPROM, which has limited writes (it's a high number, but ...)

Reply:
Well, this is a topic that is near and dear to me, actually.  I made a big PR a few months ago that cleaned up a bunch of unnecessary EEPROM writes, and added "noeeprom" versions of the functions that normally write to eeprom. 

For instance: 
```suggestion
    rgblight_enable_noeeprom();
```

You can find a list of all the commands here: 
https://docs.qmk.fm/#/feature_rgblight?id=functions

Reply:
If you can. check out how I butchered the code. This still gives different colors for layers and still the change occurs only after the second press of the layer key. which makes the thing a gimmick. Would you have an idea how to fix this?
Another issue is, without the EEPROM writes, just the connected half of my Let's Split changes color. Is there any chance on getting this without the using up the memory?